### PR TITLE
[addons] add settings support for property condition (basic conditions only)

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description
Before was it now possible to get a for Kodi defined condition on addon
settings. With this are the basic conditions also available for addon,
to define OS related parts.

The more complex conditions are not added, not sure if this goes then
to far to allow addons also this.

I'm not sure if this can in Leia or to wait for M*.

## How Has This Been Tested?
Tested with this commit here:
https://github.com/xbmc/visualization.spectrum/pull/25/commits/cfef6c42ec5e9cf957862ad4b862ce0f43ca44b8

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/addons/settings/AddonSettings.cpp [hunk 1]
```diff
@@ -391,6 +392,13 @@ void CAddonSettings::InitializeControls()
 
 void CAddonSettings::InitializeConditions()
 {
+  CSettingConditions::Initialize();
+
+  // add basic conditions
+  const std::set<std::string> &simpleConditions = CSettingConditions::GetSimpleConditions();
+  for (std::set<std::string>::const_iterator itCondition = simpleConditions.begin(); itCondition != simpleConditions.end(); ++itCondition)
+    GetSettingsManager()->AddCondition(*itCondition);
```
Comment (ksooo) [inline xbmc/addons/settings/AddonSettings.cpp]:
```c++
for (const auto& condition : simpleConditions)
    GetSettingsManager()->AddCondition(condition);
```

File: xbmc/addons/settings/AddonSettings.cpp [hunk 2]
```diff
@@ -391,6 +392,13 @@ void CAddonSettings::InitializeControls()
 
 void CAddonSettings::InitializeConditions()
 {
+  CSettingConditions::Initialize();
+
+  // add basic conditions
+  const std::set<std::string> &simpleConditions = CSettingConditions::GetSimpleConditions();
```
Comment (ksooo) [inline xbmc/addons/settings/AddonSettings.cpp]:
```c++
const std::set<std::string>& simpleConditions = ...
```
align `&`to the type, please.

**Comment by ksooo** [comment]
Could you please explain what this change is good for? What can an addon do after this change what was not possible before?

File: xbmc/addons/settings/AddonSettings.cpp (hunk 1 above)
Comment (AlwinEsch) [reply xbmc/addons/settings/AddonSettings.cpp]: Yeah, becomes changed :-D, was a c&p from Kodi's system settings place ;-)

**Comment by AlwinEsch** [comment]
See the link above, on a addon seetings.xml is a value like them, not possible:
```*.xml
        <setting id="pointsize" type="integer" label="30015">
          <default>3</default>
          <constraints>
            <minimum>1</minimum>
            <step>1</step>
            <maximum>50</maximum>
          </constraints>
          <control type="slider" format="integer">
            <formatlabel>30016</formatlabel>
          </control>
          <dependencies>
            <dependency type="enable" setting="mode" operator="is">2</dependency>
            <dependency type="visible">
              <or>
                <condition on="property" name="IsDefined">HAS_GL</condition>
                <condition on="property" name="IsDefined">HAS_GLES</condition>
              </or>
            </dependency>
          </dependencies>
        </setting>
```

For basic are the following by my system:
- 'has_ae_quality_levels'
- 'has_airplay'
- 'has_filesystem_smb'
- 'has_gl'
- 'has_upnp'
- 'has_web_server'
- 'has_zeroconf'
- 'have_lcms2'
- 'have_libva'
- 'have_libvdpau'
- 'have_x11'
- 'true'
- 'webserver_has_ssl'

This `property` values was not usable before on addon settings

**Comment by ksooo** [comment]
Ah, okay. Thanks for explaining.

**Pull request merged**